numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -426,16 +440,25 @@ UNICODE_setitem(PyObject *op, void *ov, void *vap)
     char *buffer;
 #endif
 
-    if (!PyBytes_Check(op) && !PyUnicode_Check(op) &&
-            PySequence_Check(op) && PySequence_Size(op) > 0) {
+    /* Handle case of assigning from an array scalar */
+    if (PyArray_Check(op) && PyArray_NDIM((PyArrayObject *)op) == 0) {
```

Comment:
I pushed a commit moving it into a function? what does not lend itself to it?
the return is a little odd in the flow of the function but it is the same without the function.